data/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -345,6 +352,25 @@ public LocalDateTime read(LocalDateTime reuse) {
     }
   }
 
+  private static class TimestampInt96Reader extends ParquetValueReaders.PrimitiveReader<LocalDateTime> {
+    private static final long UNIX_EPOCH_JULIAN = 2_440_588L;
+
+    private TimestampInt96Reader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public LocalDateTime read(LocalDateTime reuse) {
+      final ByteBuffer byteBuffer = column.nextBinary().toByteBuffer().order(ByteOrder.LITTLE_ENDIAN);
```

Comment:
Note for reviewers (and future me): `toByteBuffer` returns a duplicate of the internal buffer so that it is safe for uses of it to modify the buffer's position with methods like `getLong`.